include/fiction/algorithms/simulation/sidb/is_operational.hpp

```diff
@@ -53,25 +56,58 @@ enum class operational_status : uint8_t
 };
 
 /**
- * Condition which is used to decide if a layout is operational or non-operational.
+ * Parameters for the `is_operational` algorithm.
  */
-enum class operational_condition : uint8_t
+struct is_operational_params
 {
     /**
-     * Even if the I/O pins show kinks, the layout is still considered as operational.
+     * Condition to decide whether a layout is operational or non-operational.
      */
-    TOLERATE_KINKS,
+    enum class operational_condition : uint8_t
+    {
+        /**
+         * Even if the I/O pins show kinks, the layout is still considered as operational.
+         */
+        TOLERATE_KINKS,
+        /**
+         * The I/O pins are not allowed to show kinks. If kinks exist, the layout is considered as non-operational.
+         */
+        REJECT_KINKS
+    };
+
     /**
-     * The I/O pins are not allowed to show kinks. If kinks exist, the layout is considered as non-operational.
+     * Simulation method to determine if the layout is operational or non-operational. There are three possible modes:
```

Comment:
```suggestion
     * Simulation method to determine if the layout is operational or non-operational. There are three possible strategies:
```
since the `enum class` is now called `..._strategy`.

Reply:
good catch! Thanks!